connect/src/com/telenor/connect/headerenrichment/HeLogic.java

```diff
@@ -196,6 +197,9 @@ private static void handleHeTokenResult(HeTokenResponse heTokenResponse) {
         if (heTokenCallback != null) {
             heTokenCallback.done();
         }
+        if (!failedToGetToken() && !heWasExecutedAndReceivedSuccessResponse) {
```

Comment:
Don't really need to check if this was set to true previously, before setting it to true again. Might even just write `heWasExecBlahBlah = !failedToGetToken()`, but I agree the `if` test is a bit clearer.

Reply:
The logic had changed. Please, check the new version out.